csharp/src/Microsoft.ML.OnnxRuntime/SessionOptions.shared.cs

```diff
@@ -51,6 +52,8 @@ public SessionOptions()
             : base(IntPtr.Zero, true)
         {
             NativeApiStatus.VerifySuccess(NativeMethods.OrtCreateSessionOptions(out handle));
+            // Instantiate the OrtEnv singleton if not already done.
+            OrtEnv.Instance();
```

Comment:
Should we throw here instead of creating a default environment? What if the user creates an environment with options after creating a SessionOptions object? Those options won't take effect.

Reply:
If the user attempts to create an environment with options when one already exists, that API would throw.

Reply:
Ok, in that case can we throw with a helpful error message that says something to the effect of 'you've the option to create an environment of your own but you've to do so before creating SessionOptions'?

Reply:
Yes, we are doing it.